src/tiledb/cloud/geospatial/ingestion.py

```diff
@@ -0,0 +1,1287 @@
+                    )
+                else:
+                    full_extents = extents
+
+            return GeoMetadata(
+                extents=full_extents, crs=crs, paths=sources, geometry_schema=schema
+            )
+
+
+def get_raster_metadata(
+    sources: Iterable[os.PathLike],
+    *,
+    config: Mapping[str, Any] = None,
+    dst_tile_size: [int] = RASTER_TILE_SIZE,
+    res: Tuple[float, float] = None,
+    verbose: bool = False,
+    id: str = "raster_metadata",
+    trace: bool = False,
+    log_uri: Optional[str] = None,
+) -> GeoMetadata:
+    """Return geospatial metadata for a sequence of input raster data files
+
+    :param sources: iterator, paths or path to process
+    :param config: dict, configuration to pass on tiledb.VFS
+    :param dst_tile_size: int, tile size of the destination mosaic
+    :param verbose: bool, enable verbose logging, default is False
+    :param trace: bool, enable trace logging, default is False
+    :param log_uri: Optional[str] = None,
+    :Return: GeoMetadata, a populated GeoMetadata objects with blocks
+             and the files that contribute to each block
+    """
+    if len(sources) == 0:
+        raise ValueError("Input raster datasets required")
+
+    logger = get_logger_wrapper(verbose)
+    full_extents = None
+    crs = None
+    num_bands = None
+    do_res_check = True
+    dtype = None
+
+    if res:
+        xres, yres = res
+        do_res_check = False
+    else:
+        xres = None
+        yres = None
+
+    def group_by_raster_block(
+        meta: Sequence[GeoMetadata],
+    ) -> Tuple[GeoBlockMetadata, ...]:
+        nonlocal logger
+
+        # fast bulk load of geometries in to a r-tree
+        def load_geoms(meta):
+            for i, im in enumerate(meta):
+                yield (i, im.extents.bounds, None)
+
+        idx = index.Index(load_geoms(meta))
+        bounds = tuple(idx.bounds)
+        if bounds != full_extents.bounds:
+            logger.error(
+                "Index bounds %r Expected bounds %r", bounds, full_extents.bounds
+            )
+            raise ValueError("Unexpected bounds mismatch in raster ingest")
+
+        width = (bounds[2] - bounds[0]) / xres
+        height = (bounds[3] - bounds[1]) / yres
+
+        dst_transform = rasterio.transform.from_bounds(
+            *full_extents.bounds, width, height
+        )
+        dst_windows = rasterio.windows.window_split(height, width, dst_tile_size**2)
+
+        results = []
+        for _, w in dst_windows:
+            region_bounds = rasterio.windows.bounds(w, dst_transform)
+            # find intersection
+            intersects = tuple(idx.intersection(region_bounds))
+            if len(intersects) > 0:
+                # append the result in pixel coords
+                results.append(
+                    GeoBlockMetadata(
+                        ranges=w.toranges(),
+                        files=tuple([meta[s].paths for s in intersects]),
```

Comment:
With the `converter=tuple` change to GeoBlockMetadata, this can be changed to `files=meta[s].paths for s in intersects,`, without needing to be explicitly wrapped with `tuple()`.

Reply:
👍